### PR TITLE
Update TLDs, Allow Resolution Of _dnslink

### PR DESCRIPTION
**Pull request by bonedaddy**
Updates the map of valid TLDs to `Version 2019031400`. Also includes a script to help regenerate the contents of the mapping.

Additionally it enables the validation of dnslink domains

### CODE REVIEW

File: tlds.go
Comment (Stebalien) [inline tlds.go:1535]: ZW?

File: is_domain.go
Comment (Stebalien) [inline is_domain.go]: This library is a general-purpose "is domain" library so we should probably do this elsewhere if we actually need it.

**Review by Stebalien — CHANGES_REQUESTED** [review]
Thanks! I'll see if I can get write access so I can merge this (modulo my comments).

File: tlds.go
Comment (bonedaddy) [reply tlds.go:1535]: Fixed, was due to a bad copy-paste

File: is_domain.go
Comment (bonedaddy) [reply is_domain.go]: It makes sense what you're saying, however missing this chunk of code appears to have been what was causing `go-dnslink` to be unable to resolve `_dnslink.` records, as it would pass the TLD validation check, but then fail on the regexp check, presumably due to regexp statement not containing an `_`. 

My other inclination was to adjust the regexp to allow for `_` however, that would cause a larger problem I think of records like `foo.b_ar.com` being valid.

However that being said, I'm not opposed to removing this, but I think removing it would lead to some extra complexity and verbose handling of strings in libraries like `go-dnslink` which would then require multiple repeats of the trimming of `_dnslink.` so that this library wouldn't fail on this particular check.

**Comment by Kubuxu** [comment]
@Stebalien should we adopt this repo under `ipfs`. We already diverged from original in `gxed` org: https://github.com/gxed/go-is-domain

File: is_domain.go
Comment (bonedaddy) [reply is_domain.go]: Actually after thinking about this more you are indeed correct, will remove.

**Comment by Stebalien** [comment]
@Kubuxu I'm fine leaving it where it is, really. It's Juan's package.

**Review by Stebalien — APPROVED** [review]

**Pull request merged**
